core/object/make_virtuals.py

```diff
@@ -16,15 +16,26 @@
 		}    \\
 	}\\
     if (unlikely(_get_extension() && !_gdvirtual_##m_name##_initialized)) {\\
-        /* TODO: C-style cast because GDExtensionStringNamePtr's const qualifier is broken (see https://github.com/godotengine/godot/pull/67751) */\\
-        _gdvirtual_##m_name = (_get_extension() && _get_extension()->get_virtual) ? _get_extension()->get_virtual(_get_extension()->class_userdata, (GDExtensionStringNamePtr)&_gdvirtual_##m_name##_sn) : (GDExtensionClassCallVirtual) nullptr;\\
+        _gdvirtual_##m_name##_data = nullptr;\\
+        if (_get_extension()->get_virtual_call_data) {\\
+            _gdvirtual_##m_name##_data = _get_extension()->get_virtual_call_data(_get_extension()->class_userdata, &_gdvirtual_##m_name##_sn);\\
+        } else if (_get_extension()->get_virtual) {\\
+            _gdvirtual_##m_name##_data = (void *)_get_extension()->get_virtual(_get_extension()->class_userdata, &_gdvirtual_##m_name##_sn);\\
+        }\\
         _gdvirtual_##m_name##_initialized = true;\\
     }\\
-	if (_gdvirtual_##m_name) {\\
+	if (_gdvirtual_##m_name##_data) {\\
 		$CALLPTRARGS\\
 		$CALLPTRRETDEF\\
-		_gdvirtual_##m_name(_get_extension_instance(),$CALLPTRARGPASS,$CALLPTRRETPASS);\\
-		$CALLPTRRET\\
+        if (_get_extension()->get_virtual_call_data) {\\
+            if (_get_extension()->call_virtual_with_data) {\\
```

Comment:
In this path the method may never be called yet the method returns true, meaning the return value is uninitialised

Reply:
👍 Good catch. I'll have to think about how to fix without triggering clang linters.

Reply:
Probably good indication to go over the code to ensure it does what it is supposed to do, good that the CI catches unassigned data related things, though it's very vague in the errors